lib/http/fetch.js

```diff
@@ -0,0 +1,159 @@
+var AWS = require('../core');
+var pipeThrough = AWS.util.pipeThrough;
+var TransformStream = typeof TransformStream !== 'undefined' ? TransformStream :
+  AWS.util.stream && AWS.util.stream.TransformStream ? AWS.util.stream.TransformStream :
+  typeof WebStreamsPolyfill !== 'undefined' ? WebStreamsPolyfill.TransformStream :
+  undefined;
+require('../http');
+
+/**
+ * @api private
+ */
+AWS.FetchClient = AWS.util.inherit({
+  handleRequest: function handleRequest(httpRequest, httpOptions, callback, errCallback) {
+    var self = this;
+    var endpoint = httpRequest.endpoint;
+    if (!httpOptions) httpOptions = {};
+
+    var emitter = new EventEmitter();
+
+    // Construct href
+    var href = endpoint.protocol + '//' + endpoint.hostname;
+    if (endpoint.port !== 80 && endpoint.port !== 443) {
+      href += ':' + endpoint.port;
+    }
+    href += httpRequest.path;
+
+    // Construct headers
+    var headers = new Headers();
+    AWS.util.each(httpRequest.headers, function (key, value) {
+      // See: https://fetch.spec.whatwg.org/#forbidden-header-name
+      if (key !== 'Content-Length' && key !== 'Expect' && key !== 'Host') {
+        headers.set(key, value);
+      }
+    });
+
+    // Send cookies?
+    var credentials = 'omit';
+    if (httpOptions.xhrWithCredentials) {
+      credentials = 'include';
+    }
+
+    // Construct request
+    // See: https://developer.mozilla.org/en-US/docs/Web/API/Request/Request
+    var req = {
+      method: httpRequest.method,
+      headers: headers,
+      credentials: credentials
+    };
+    if (httpRequest.body) req.body = httpRequest.body;
+    var request = new Request(href, req);
+
+    // Send request
+    fetch(request).then(function(response) {
+        if (!response.ok) throw Error(response.statusText);
+        return response;
+    }).then(function(response) {
+      emitter.statusCode = response.status;
+      emitter.headers = self.parseHeaders(response.headers);
+      emitter.abort = function() { response.body.cancel(); };
```

Comment:
I believe you should be using [AbortSignal](https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal) to abort a request/response

Reply:
Yes indeed. 😄 I'm just concerned on the level of browser support for AbortSignal. Need research as the pollyfills I've seen so far don't do it correctly: https://www.npmjs.com/package/abortcontroller-polyfill